transform.go

```diff
@@ -103,11 +103,19 @@ func transformImage(m image.Image, opt Options) image.Image {
 
 	// never resize larger than the original image
 	if !opt.ScaleUp {
-		if w > imgW {
-			w = imgW
+		if w >= imgW {
+			if h == 0 {
```

Comment:
You're right, the patch is wrong. But I think the correct condition would be:
```go
// if requested width and height match the original, skip resizing
if (w == imgW && (h == imgH || h == 0)) || (w==0 && h == imgH) {
    w, h = 0, 0
}
```

It needs to handle the case where only one of the dimentions to scale is set.